editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/AdvancedBeanPropertiesSection.java

```diff
@@ -0,0 +1,300 @@
+import org.fusesource.ide.camel.model.service.core.util.CamelComponentUtils;
+import org.fusesource.ide.foundation.core.util.Strings;
+
+/**
+ * @author bfitzpat
+ */
+public class AdvancedBeanPropertiesSection extends FusePropertySection {
+
+	/**
+	 * 
+	 * @param folder
+	 */
+	@Override
+	protected void createContentTabs(CTabFolder folder) {
+		List<Parameter> props = new ArrayList<>();
+		List<String> tabsToCreate = new ArrayList<>();
+		tabsToCreate.add(GROUP_COMMON);
+		Parameter idParam = BeanConfigUtil.createParameter(CamelBean.PROP_ID, String.class.getName());
+		idParam.setRequired("true");
+		props.add(idParam);
+		Parameter classParam = BeanConfigUtil.createParameter(CamelBean.PROP_CLASS, String.class.getName());
+		classParam.setRequired("true");
+		props.add(classParam);
+		props.add(BeanConfigUtil.createParameter(CamelBean.PROP_SCOPE, String.class.getName()));
+		props.add(BeanConfigUtil.createParameter(CamelBean.PROP_DEPENDS_ON, String.class.getName()));
+		props.add(BeanConfigUtil.createParameter(CamelBean.PROP_INIT_METHOD, String.class.getName()));
+		props.add(BeanConfigUtil.createParameter(CamelBean.PROP_DESTROY_METHOD, String.class.getName()));
+
+		props.sort(new ParameterPriorityComparator());
+
+		for (String group : tabsToCreate) {
+			CTabItem contentTab = new CTabItem(this.tabFolder, SWT.NONE);
+			contentTab.setText(Strings.humanize(group));
+
+			Composite page = this.toolkit.createComposite(folder);
+			page.setLayout(new GridLayout(4, false));
+
+			generateTabContents(props, page);
+
+			contentTab.setControl(page);
+
+			this.tabs.add(contentTab);
+		}
+	}
+
+	private IJavaProject getJavaProject() {
+		IProject project = CamelUtils.project();
+		if (project != null) {
+			IJavaProject jproject = JavaCore.create(project);
+			if (jproject.exists()) {
+				return jproject;
+			}
+		}
+		return null;
+	}
+
+	private IStatus validateMethod(Object value) {
+		Object control = modelMap.get(CamelBean.PROP_CLASS);
+		if (control != null) {
+			String className = (String) control;
+			String methodName = (String) value;
+			if (!Strings.isEmpty(methodName)) {
+				IJavaProject jproject = getJavaProject();
+				if (jproject != null) {
+					try {
+						IType foundClass = jproject.findType(className);
+						boolean foundMethod = BeanConfigUtil.hasMethod(methodName, foundClass);
+						if (!foundMethod) {
+							return ValidationStatus
+									.error("Method " + methodName + " must exist in class " + className + ".");
+						} else {
+							return ValidationStatus.ok();
+						}
+					} catch (JavaModelException e) {
+						CamelEditorUIActivator.pluginLog().logError("Error validating method for class " + className,
+								e);
+					}
+				}
+			}
+		}
+		return ValidationStatus.ok();
+	}
+
+	/**
+	 * 
+	 * @param props
+	 * @param page
+	 * @param ignorePathProperties
+	 * @param group
+	 */
+	protected void generateTabContents(List<Parameter> props, final Composite page) {
```

Comment:
What do you think to extract a method called with something like createPropertyFieldEdition which ends up with something like

`protected void generateTabContents(List<Parameter> props, final Composite page) {
		props.sort(new ParameterPriorityComparator());
		for (Parameter p : props) {
			final Parameter prop = p;
			createPropertyLabel(toolkit, page, p);
			createPropertyFieldEdition(page, p, prop);
		}
	}

	protected void createPropertyFieldEdition(final Composite page, Parameter p, final Parameter prop) {
		ISWTObservableValue uiObservable = null;
		IObservableValue<Object> modelObservable;
		IValidator validator = null;


		AbstractTextFieldParameterPropertyUICreator txtFieldCreator = null;

		if (p.getName().contentEquals(CamelBean.PROP_CLASS)) {
			txtFieldCreator = createTextFieldWithClassBrowse(p, page);
			validator = new BeanClassExistsValidator();
		} else if (p.getName().contentEquals(CamelBean.PROP_INIT_METHOD)
				|| p.getName().contentEquals(CamelBean.PROP_DESTROY_METHOD)) {
			txtFieldCreator = createTextFieldWithMethodBrowse(p, page);
			validator = this::validateMethod;
		} else if (p.getName().contentEquals(CamelBean.PROP_ID)) {
			txtFieldCreator = new AttributeTextFieldPropertyUICreator(dbc, modelMap, eip, selectedEP, p, page,
					getWidgetFactory());
			txtFieldCreator.create();
			validator = new NewBeanIdPropertyValidator(p, selectedEP);
		} else if (CamelComponentUtils.isTextProperty(prop) || CamelComponentUtils.isCharProperty(prop)) {
			txtFieldCreator = new AttributeTextFieldPropertyUICreator(dbc, modelMap, eip, selectedEP, p, page,
					getWidgetFactory());
			txtFieldCreator.create();

			if (p.getRequired() != null && "true".contentEquals(p.getRequired())) {
				validator = new PropertyRequiredValidator(p);
			}

			// UNSUPPORTED PROPERTIES / REFS
		} else if (CamelComponentUtils.isUnsupportedProperty(prop)) {
			new UnsupportedParameterPropertyUICreatorForAdvanced(dbc, modelMap, eip, selectedEP, p, page,
					getWidgetFactory()).create();
		}

		if (txtFieldCreator != null) {
			// initialize the map entry
			modelMap.put(p.getName(), txtFieldCreator.getControl().getText());
			// create observables for the control
			uiObservable = txtFieldCreator.getUiObservable();
		}

		if (uiObservable != null) {
			// create UpdateValueStrategy and assign to the binding
			UpdateValueStrategy strategy = new UpdateValueStrategy();
			strategy.setBeforeSetValidator(validator);

			// create observables for the Map entries
			modelObservable = Observables.observeMapEntry(modelMap, p.getName());
			// bind the observables
			Binding bindValue = dbc.bindValue(uiObservable, modelObservable, strategy, null);
			ControlDecorationSupport.create(bindValue, SWT.TOP | SWT.LEFT);
		}
	}`

after that we need to understand why the parameter prop is introduced, seems not useful but there is a strange "final" which leads to think that it has a very specific reason (that i wasn't able to find at first sight)

we still have on quite big method (but under the cyclomatic complexity of 15).
Maybe we can also extract the last if statement to something like bindPropertyFieldEdition